### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by morethanreal**
I'm gonna open a PR before this gets too crazy. PTAL @cdata (sorry!) cc @notwaldorf @addyosmani @robdodson

Doc here: https://docs.google.com/document/d/1j6W_KuOS3khaOLS4Uw6g85W4CpsLwQGjvpkGl2PbCOo/edit#

Live demos:
http://morethanreal.github.io/neon-animated-pages-demo/index.html

### CODE REVIEW

**Comment by cdata** [comment]
image

:'(

File: animations/cascade-slide-up-animation.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  Polymer({
+
+    is: 'cascade-slide-up-animation',
+
+    behaviors: [
+      Polymer.NeonAnimationBehavior
+    ],
+
+    configure: function(config) {
```
Comment (cdata) [inline animations/cascade-slide-up-animation.html]: I'm pretty sure `configure` is gone now. It's gone now, right?

File: animations/fade-in-animation.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  Polymer({
+
+    is: 'fade-in-animation',
+
+    behaviors: [
+      Polymer.NeonAnimationBehavior
+    ],
+
+    configure: function(config) {
```
Comment (cdata) [inline animations/fade-in-animation.html]: Ditto, re: https://github.com/PolymerElements/neon-animated-pages/pull/3/files#r30198477

File: animations/fade-out-animation.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  Polymer({
+
+    is: 'fade-out-animation',
+
+    behaviors: [
+      Polymer.NeonAnimationBehavior
+    ],
+
+    configure: function(config) {
```
Comment (cdata) [inline animations/fade-out-animation.html]: Ditto, re: https://github.com/PolymerElements/neon-animated-pages/pull/3/files#r30198477

File: animations/hero-animation.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  Polymer({
+
+    is: 'hero-animation',
+
+    behaviors: [
+      Polymer.NeonSharedElementAnimationBehavior
+    ],
+
+    configure: function(config) {
```
Comment (cdata) [inline animations/hero-animation.html]: Ditto, re: https://github.com/PolymerElements/neon-animated-pages/pull/3/files#r30198477

File: neon-animation-runner-behavior.html [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+<script>
+
+  Polymer.NeonAnimationRunnerBehavior = {
+
+    properties: {
+
+      _animationMeta: {
+        type: Object,
+        value: function() {
+          return new Polymer.IronMeta({type: 'animation'});
```
Comment (cdata) [inline neon-animation-runner-behavior.html]: There is a new thing from @sjmiles for reading stored meta called `IronMetaQuery`. Maybe look into using that?

**Comment by cdata** [comment]
A lot of this stuff seems new. If you could provide a brief, one-sentence description for each behavior / element in a leading comment, it would lend context and make the code easier to review and kick start documentation as well.

**Comment by morethanreal** [comment]
`configure` is my own API. Since it's not a `Polymer.Base` method, it should be safe with that name?

**Comment by morethanreal** [comment]
Yes, this needs a lot of docs. I've been keeping track of them in that Google doc, and will paste them in when it's ready for merging.

**Pull request merged**